server/vulnerabilities/nvd/tools/cvefeed/nvd/match_cpe.go

```diff
@@ -69,13 +69,42 @@ func (cm *cpeMatch) Match(attrs []*wfn.Attributes, requireVersion bool) (matches
 		if cm.match(attr, requireVersion) {
 			matches = append(matches, attr)
 		}
-		if osMatch := cm.MatchTargetSW(attr); osMatch != nil {
+		if osMatch := cm.matchTargetSW(attr); osMatch != nil {
 			matches = append(matches, osMatch)
 		}
 	}
 	return matches
 }
 
+// matchTargetSW returns an OS CPE for the given application CPE
+// if the application CPE has a target software attribute and
+// matches the given cpeMatch
+func (cm *cpeMatch) matchTargetSW(attr *wfn.Attributes) *wfn.Attributes {
+	if cm == nil || attr == nil {
+		return nil
+	}
+
+	if attr.Part != "a" || attr.TargetSW == "" {
+		return nil
+	}
+
+	osAttr := &wfn.Attributes{
+		Part:    "o",
+		Product: attr.TargetSW,
+	}
+
+	partMatches := cm.Part == osAttr.Part
+	productMatches := cm.Product == osAttr.Product
+	versionMatches := cm.Version == wfn.NA || cm.Version == wfn.Any
```

Comment:
Should version be considered matched if version is `NA`?

Reply:
the use of `NA` in CPE definitions are still a bit vague for me, but I'm seeing NA fields in NVD:  https://nvd.nist.gov/vuln/detail/CVE-2023-44325 (`cpe:2.3:o:apple:macos:-:*:*:*:*:*:*:*`)